tableau_utilities/scripts/connection.py

```diff
@@ -0,0 +1,112 @@
+import os
+import sys
+
+
+import tableau_utilities as tu
+from tableau_utilities.tableau_server.tableau_server import TableauServer
+
+
+def connection_settings(args):
+    """ Creates a dictionary with named connection information
+
+    Returns:
+        connection_settings: Dictionary with the connection information
+    """
+
+    conn_user = args.conn_user
+    conn_pw = args.conn_pw
+    conn_type = args.conn_type
+    conn_db = args.conn_db
+    conn_schema = args.conn_schema
+    conn_host = args.conn_host
+    conn_role = args.conn_role
+    conn_warehouse = args.conn_warehouse
+
+    if args.conn == 'settings_yaml':
+        conn_user = settings['embed_connection']['username']
+        conn_pw = settings['embed_connection']['password']
+        conn_type = settings['embed_connection']['class_name']
+        conn_db = settings['embed_connection']['dbname']
+        conn_schema = settings['embed_connection']['schema']
+        conn_host = settings['embed_connection']['server']
+        conn_role = settings['embed_connection']['service']
+        conn_warehouse = settings['embed_connection']['warehouse']
+    elif args.conn == 'os_env':
+        conn_user = os.getenv("CONN_USERNAME")
+        conn_pw = os.getenv("CONN_PASSWORD")
+        conn_type = os.getenv("CONN_TYPE")
+        conn_db = os.getenv("CONN_DB")
+        conn_schema = os.getenv("CONN_SCHEMA")
+        conn_host = os.getenv("CONN_HOST")
+        conn_role = os.getenv("CONN_ROLE")
+        conn_warehouse = os.getenv("CONN_WAREHOUSE")
+
+
+    connection_settings = {'conn_user': conn_user,
+                           'conn_pw': conn_pw,
+                           'conn_type': conn_type,
+                           'conn_db': conn_db,
+                           'conn_schema': conn_schema,
+                           'conn_host': conn_host,
+                           'conn_role': conn_role,
+                           'conn_warehouse': conn_warehouse
+                           }
+
+    return connection_settings
+
+
+def update_connection(datasource, conn_settings):
+    """
+
+    Args:
+        datasource: The datasource object
+        conn_settings: The connection setttings dictionary
+
+    """
+    connection_type = conn_settings['conn_type']
+    named_conn = datasource.connection.named_connections.get(conn_settings['conn_type'])
+
+    # named_conn = conn_settings['conn_type']
+    print(named_conn)
+
+    named_conn.connection.class_name = connection_type
+    named_conn.connection.server = conn_settings['conn_host']
+    named_conn.connection.username = conn_settings['conn_user']
+    named_conn.connection.service = conn_settings['conn_role']
+    named_conn.connection.dbname = conn_settings['conn_db']
+    named_conn.connection.schema = conn_settings['conn_schema']
+    named_conn.connection.warehouse = conn_settings['conn_warehouse']
+
+    datasource.connection.named_connections[connection_type].connection = named_conn
```

Comment:
```suggestion
    connection_type = conn_settings['conn_type']
    
    datasource.connection.named_connections[connection_type].connection.server = conn_settings['conn_host']
    datasource.connection.named_connections[connection_type].connection.username = conn_settings['conn_user']
    datasource.connection.named_connections[connection_type].connection.service = conn_settings['conn_role']
    datasource.connection.named_connections[connection_type].connection.dbname = conn_settings['conn_db']
    datasource.connection.named_connections[connection_type].connection.schema = conn_settings['conn_schema']
    datasource.connection.named_connections[connection_type].connection.warehouse = conn_settings['conn_warehouse']
```

`datasource.connection.named_connections[connection_type].connection = named_conn`
This line was assigning a named connection to a connection of a named connection, which I believe was causing the recursive error you got.
But this can be solved by just updating the connection attributes directly.

Reply:
Also also! The ParentConnection class does have an `update` function, if you wanted to keep you current code but simplify line 80 to this:
`datasource.connection.update(named_conn.connection)`

Reply:
@JustinGrilli I took your code and that worked great.  Thanks!  

I'm optimistic that I'll finish the CLI this weekend. 🤞